### DISCUSSION TITLE
variable_format for write_xport SAS

### DISCUSSION THREAD

**Discussion by raghavsharda** [Roche/pyreadstat#137] [Q&A] [answered]
Hi 

I have CSV data which I am trying to convert to XPT.
I know the SAS format which is provided to me for those columns.
I am getting confused as to how do i assign those 4 columns to SAS formats
```

# what i have right now -- not sure if this is even slightly correct
formats = {'ID':str, 'DATE':date,'COM':str,'RESLT':str}

# what i am trying to actually do i.e make sure DATE columns is of SAS format DATE9
formats = {'ID':'$11', 'DATE':'DATE9','COM':'CHAR/150','RESLT':'CHAR/13'}

pyreadstat.write_xport(df_write, path_ouput , file_label=file_label, column_labels=column_labels, variable_format = formats )

```

How can I specify variable format given I know the SAS format?

**Comment by ofajardo** [comment, marked as answer]
try passing the second dictionary i.e:
```
formats = {'ID':'$11', 'DATE':'DATE9','COM':'CHAR/150','RESLT':'CHAR/13'}
```
I am not sure about CHAR and $11 but I think DATE9 should work. 

However, consider these are SAS formats, I am not sure if these work in XPT as well.

**Reply by ofajardo** [reply]
bottom line is formats should be a dict where they key is the variable name and the value is the format in the application in this case the SAS format you want to force.
